### PR TITLE
This should fix issue 141

### LINKED ISSUE #141: Wikidata Plugin Error
For some tracks wikidata plugin freezes Picard. Here's the debug log entry.

```
I: 15:28:35 WIKIDATA: fetching the folowing url wikidata.org/wiki/Special:EntityData/Q640529.rdf
Traceback (most recent call last):
  File "/usr/lib/picard/picard/webservice.py", line 315, in _process_reply
    self._handle_reply(reply, request, handler, xml, refresh)
  File "/usr/lib/picard/picard/webservice.py", line 304, in _handle_reply
    handler(document, reply, error)
  File "/home/dimon3000/.config/MusicBrainz/Picard/plugins/wikidata.zip/wikidata.py", line 196, in parse_wikidata_response
KeyError: u'fc2755ef-937d-3b95-8977-a4f64bd80243'
```

### PR DESCRIPTION
**Pull request by dns53**
There are multiple wikidata item so a request is made for each item.
When each request is done it is removing itself from the list of outstanding requests.
removing this delete stops it from crashing.
This should hopefully fix #141

### CODE REVIEW

File: plugins/wikidata/wikidata.py [hunk 1]
```diff
@@ -122,17 +119,18 @@ def musicbrainz_release_lookup(self, item_id, metadata, response, reply, error):
...
-            for tagger in self.taggers[item_id]:
-                tagger._requests -= 1
-                if tagger._requests == 0:
-                    tagger._finalize_loading(None)
-                log.debug('WIKIDATA:  TOTAL REMAINING REQUESTS %s' %
-                          tagger._requests)
-            del self.requests[item_id]
-            self.lock.release()
-
+        self.lock.acquire()
```
Comment (sambhav) [inline plugins/wikidata/wikidata.py]: You can use the lock as a context manager.
Via
`with self.lock:`
and remove the acquire and release parts

Comment (dns53) [reply plugins/wikidata/wikidata.py]: Happy to have it changed.
As pull  #142l is also cleaning things up i should probably merge with that as well.

Comment (sambhav) [reply plugins/wikidata/wikidata.py]: Oh, yes, didn't notice that PR.

**Comment by mineo** [comment]
@dns53 I've merged #142 into master and did a test rebase of your changes on top of that. The result is https://github.com/mineo/picard-plugins/commit/374cd895f8c615228d39d57038896865b7930a8f. You should be able to apply that on your branch through the following:

```shell
# Note the output of this if anything goes wrong, should be 40df080e9bb131511e1fab5895698fc3f3c6c32a
git rev-parse HEAD
# Assuming origin is the metabrainz remote
git fetch origin
git reset --hard origin/2.0
git am am https://github.com/mineo/picard-plugins/commit/374cd895f8c615228d39d57038896865b7930a8f.patch
git push --force
```

If, after `git rev-parse HEAD` anything goes wrong, just reset your branch back to the current commit with `git reset --hard 40df080e9bb131511e1fab5895698fc3f3c6c32a`.

**Comment by faminebadger** [comment]
This won't work and has errors in it.

See https://github.com/metabrainz/picard-plugins/issues/141#issuecomment-393971285

**Comment by zas** [comment]
Status ?

**Comment by dns53** [comment]
Hmm Looks like we all forgot about this.
Moneo has some good cleanup commits and I have a few commits that should fix the bugs but we have not merged the two.

**Comment by phw** [comment]
@dns53 Any chance to get this PR updated?

**Comment by zas** [comment]
Staled PR

**Pull request closed without merging**